security/limits.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh -e
+
+PAM_UID=$(getent passwd "${PAM_USER}" | cut -d: -f3)
+
+if [ "${PAM_UID}" -ge 1000 ]; then
+    /usr/bin/systemctl set-property "user-${PAM_UID}.slice" \
+                       CPUQuota=50% MemoryLimit=512M BlockIOWeight=10
```

Comment:
`CPUQuota=50%` seems kind of high for the amount of users. Is there a good resource to read up on how this would work? (is this 50% per cpu per user?)

Reply:
50% of one core, max.  The documentation is in the `systemd.resource-control(5)` manpage.

The 50% value is kinda pulle out of my hat, but it's high enough not to make the sessions useless (for instance when compiling things) and low enough to prevent people from running silly things like cryptocurrency miners (also, prevent them from interfering too badly with other users and deamons)

Reply:
@DeviaVir - this means that no user can run more than 50% of a single core at all times. This doesn't mean each user gets a dedicated core, but instead that you'll need quite a few users to even attempt maxing it out.